.github/workflows/java-cd.yml

```
@@ -244,22 +238,46 @@ jobs:
               uses: ./.github/workflows/install-shared-dependencies
               with:
                   os: ${{ matrix.host.OS }}
+                  engine-version: "7.2.5"
                   target: ${{ matrix.host.TARGET }}
                   github-token: ${{ secrets.GITHUB_TOKEN }}
+                  use-cache: false
 
             - name: Install protoc (protobuf)
               uses: arduino/setup-protoc@v3
               with:
                   version: "26.1"
                   repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+            - name: Find available port
+              id: port
+              run: |
+                  if ${{ matrix.host.OS == 'ubuntu'}}; then
+                      sudo apt install net-tools
+                      while
+                        port=$(shuf -n 1 -i 40000-65000)
+                        netstat -atun | grep -q "$port"
+                      do
+                        continue
+                      done
+                  else
+                    export port=`comm -23 <(seq 49152 65535 | sort) <(awk '{print $4}' | cut -d':' -f2 | sort -u) | shuf | head -n 1`
+                  fi
+
+                  echo "PORT=$port" >> $GITHUB_OUTPUT
+
+            - name: Start Valkey server
+              run:
+                  cd ~/valkey/src
+                  ./valkey-server --port ${{ steps.port.outputs.PORT }}&
```

```suggestion
                  valkey-server --port ${{ steps.port.outputs.PORT }} &
```
`valkey-server` should be in `$PATH` if installed properly